templates/community/events.macros.j2

```diff
@@ -0,0 +1,314 @@
+    <section class="splash-talks extend-full">
+      {{ content.divider(title='Featured Talks & Workshops') }}
+      <div class="splash-section">
+        {% for talk in talks %}
+          <article class="splash-talk">
+            <h3 class="splash-title">
+              {{ utility.link_if(
+                url=content.page_link(talk),
+                content=talk.title|typogrify
+                ) }}
+            </h3>
+            <div class="splash-brag">
+              {% set brag = talk.config.brag %}
+              {% set brag = talk.render_summary() if (brag == 'summary') else brag|rst %}
+              {{ brag }}
+            </div>
+          </article>
+        {% endfor %}
+      </div>
+    </section>
+  {% endif %}
+{% endmacro %}
+
+
+{% macro videos(
+  talks=none,
+  slice=10
+) %}
+  {% set events = talks|collect('events', label_with='slug') %}
+  {% set events = events|selectattr('feature')|sort(attribute='date')|reverse|list %}
+  {% set title = 'Featured Videos' if (slice > 1) else 'Featured Video' %}
+
+  {% if events|length %}
+    <section class="talk-videos">
+      {{ content.divider(title=title) }}
+
+      <figure class='video-gallery extend-large'>
+        {% for event in events %}
+          {% if (loop.index <= slice) %}
+            <div class="video-item">
+              <div class="video">
+                {{ event.video|safe }}
+              </div>
+            </div>
+          {% endif %}
+        {% endfor %}
+      </figure>
+    </section>
+  {% endif %}
+{% endmacro %}
+
+
+{% macro feature(
+  event,
+  talk=none,
+  show_author=false
+) %}
+  {% if event %}
+    <article class="feature-event">
+      {% set talk = talk if talk else all_pages|filter_pages('slug', 'eq', event.slug)|get_page %}
+      {% set talk_url = content.page_link(talk) if talk else none %}
+      {% set talk_url = talk_url if (talk_url != '/talks/tbd/') else none %}
+
+      <h3 class="event-title">
+        {% if talk %}
+          {{ utility.link_if(
+            content=talk.title|typogrify,
+            url=talk_url,
+            class='event-talk'
+            ) }}
+          <span class="meta-preposition">at</span>
+        {% endif %}
+        {{ utility.link_if(
+          content=event.venue,
+          url=event.url,
+          class='event-venue'
+          ) }}
+      </h3>
+
+      {{ meta(
+        event=event,
+        talk=talk,
+        show_author=show_author
+        ) }}
+
+      <div class="event-summary">
+        {{ talk.render_summary() }}
+      </div>
+    </article>
+  {% endif %}
+{% endmacro %}
+
+
+{% macro list(
+  events,
+  title,
+  talk=none,
+  show_author=false,
+  details=none,
+  slice=none
+) %}
+  <section data-module="events-list">
+    {% if events %}
+      <h3 class="events-list-title">{{ title }}</h3>
+
+      {% if slice and (events|length > slice) %}
+        <button class="btn small-btn more-events" {{ utility.toggle_button('more-events') }}>
+          show all events
```

Comment:
if we are collapsing the other speaking events, I would like to suggest we keep the button under the last event, aligned left, that we are showing. I think it can be easily missed if off to the right when event names are often short.

Reply:
I went back and forth on this. What happens when you open the list? Does it move down to stay at the bottom (making it hard to close without scrolling in some cases), or does it stay in place, breaking up the list?